crates/goose-cli/src/commands/session.rs

```diff
@@ -36,20 +36,28 @@ pub async fn build_session(name: Option<String>, resume: bool) -> Session<'stati
     // Setup extensions for the agent
     for (name, extension) in ExtensionManager::get_all().expect("should load extensions") {
         if extension.enabled {
-            agent
-                .add_extension(extension.config.clone())
-                .await
-                .unwrap_or_else(|e| {
-                    let err = match e {
-                        ExtensionError::Transport(McpClientError::StdioProcessError(inner)) => {
-                            inner
-                        }
-                        _ => e.to_string(),
-                    };
-                    println!("Failed to start extension: {}, {:?}", name, err);
-                    println!("Please check extension configuration for {}.", name);
-                    process::exit(1);
-                });
+            let mut config = extension.config.clone();
+            // Ensure extension name matches the key
```

Comment:
similar to other comment, i think we just take the name from the extension and update extension manager to not return it

Reply:
well it affects some other parts negatively